app/main.py

```diff
@@ -1,6 +1,18 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    data = {}
+
+    def inner(*args: Any) -> Callable:
+        key = (func.__name__, args)
```

Comment:
No need to store func__name__ as key, remove it

Reply:
If they are two different functions but their arguments are the same, how can we distinguish the result for them? Or I don't understand something?

Reply:
Is data will one for all function really? As for me we will create for each function new data dict